### ISSUE TITLE
 TSV reader fails to read back record with string+array

### ISSUE THREAD

**Issue by philrz** [brimdata/zed#4918]
## tl;dr

With these examples we fail to read our own TSV output back as TSV input.

```
$ echo '{foo: "hello,world", bar: ["baz"]}' | zq -f tsv - | zq -i tsv -
stdio:stdin: parse error on line 2, column 7: bare " in non-quoted-field

$ echo '{foo: "helloworld", bar: ["baz"]}' | zq -f tsv - | zq -i tsv -
stdio:stdin: record on line 2: wrong number of fields
```

## Details

Repro is with Zed commit 4ae15b0.

The CSV reader doesn't seem to have the problem.

```
$ zq -version
Version: v1.11.1-17-g4ae15b03

$ echo '{foo: "hello,world", bar: ["baz"]}' | zq -f csv - | zq -i csv -
{foo:"hello,world",bar:"[\"baz\"]"}

$ echo '{foo: "helloworld", bar: ["baz"]}' | zq -f csv - | zq -i csv -
{foo:"helloworld",bar:"[\"baz\"]"}
```

It's understood that even in the CSV case `bar` gets read back as a string rather than an array. But I accept that there's going to be complexity with some types that simple won't be reversible in these primitive formats. I'm just reacting to the parse error since it means the user doesn't get to see any data.

If you're curious about the workflow that's coming up with bugs like this and #4911, what I stumbled onto during verification was trying `zq -f tsv 'fuse' sample.zng | zq -i tsv -` with the [sample.zng](https://github.com/brimdata/zui/blob/main/apps/zui/src/test/shared/data/sample.zng) test data from the Zui repo. It seems like an abusive thing to do but it does happen to boomerang without complaint in CSV format so I figured I'd watch the same with TSV. When I've seen it fail I've trimmed down the test data to the simpler repros like this and the one from #4911 as I figured that was kinder than just dumping the massive repro on you with the hundreds of fields. But if you're concerned there'll be another round of whack-a-mole feel free to try running a test with that bigger file. And when there's a PR up with a fix for this new issue I'll be sure to run it myself on the branch before we merge.

**Comment by philrz** [comment]
Verified in Zed commit 6ff0586.

The repro examples shown above now work without error,

```
$ zq -version
Version: v1.11.1-21-g6ff05865

$ echo '{foo: "hello,world", bar: ["baz"]}' | zq -f tsv - | zq -i tsv -
{foo:"hello,world",bar:"[\"baz\"]"}

$ echo '{foo: "helloworld", bar: ["baz"]}' | zq -f tsv - | zq -i tsv -
{foo:"helloworld",bar:"[\"baz\"]"}
```

Also, the larger/fused `sample.zng` test data can now be output in full as TSV without complaint.

```
$ zq -f tsv 'fuse' sample.zng | zq -i tsv - > /dev/null

$ echo $?
0
```

Thanks @mattnibs!
